src/main/java/com/o19s/es/ltr/action/TransportFeatureStoreAction.java

```diff
@@ -0,0 +1,138 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.action;
+
+import com.o19s.es.ltr.action.ClearCachesAction.ClearCachesNodesRequest;
+import com.o19s.es.ltr.action.FeatureStoreAction.FeatureStoreRequest;
+import com.o19s.es.ltr.action.FeatureStoreAction.FeatureStoreResponse;
+import com.o19s.es.ltr.feature.store.StorableElement;
+import com.o19s.es.ltr.feature.store.StoredFeature;
+import com.o19s.es.ltr.feature.store.StoredFeatureSet;
+import com.o19s.es.ltr.feature.store.StoredLtrModel;
+import com.o19s.es.ltr.feature.store.index.IndexFeatureStore;
+import com.o19s.es.ltr.ranker.parser.LtrRankerParserFactory;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.index.IndexAction;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public class TransportFeatureStoreAction extends HandledTransportAction<FeatureStoreRequest, FeatureStoreResponse> {
+    private final LtrRankerParserFactory factory;
+    private final ClusterService clusterService;
+    private final TransportClearCachesAction clearCachesAction;
+    private final Client client;
+
+    @Inject
+    public TransportFeatureStoreAction(Settings settings, ThreadPool threadPool, TransportService transportService,
+                                       ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver,
+                                       ClusterService clusterService, Client client,
+                                       LtrRankerParserFactory factory,
+                                       TransportClearCachesAction clearCachesAction) {
+        super(settings, FeatureStoreAction.NAME, false, threadPool, transportService, actionFilters,
+                indexNameExpressionResolver, FeatureStoreRequest::new);
+        this.factory = factory;
+        this.clusterService = clusterService;
+        this.clearCachesAction = clearCachesAction;
+        this.client = client;
+    }
+
+    @Override
+    protected void doExecute(FeatureStoreRequest request, ActionListener<FeatureStoreResponse> listener) {
+        logger.warn("attempt to execute a TransportFeatureStoreAction without a task");
```

Comment:
logging AND throwing exception is usually redundant. I would expect the caller to log exception as well, so we will probably end up with duplicate log messages (well, the messages will be slightly different, but the point is that we could have all the context in the exception and remove the logger.warn() call). That same pattern is used in a few other places.